frame/support/src/dispatch.rs

```diff
@@ -354,6 +362,12 @@ where
 			error: self.into(),
 		}
 	}
+	fn with_pays_fee(self, pays_fee: Pays) -> DispatchErrorWithPostInfo {
+		DispatchErrorWithPostInfo {
+			post_info: PostDispatchInfo { actual_weight: None, pays_fee },
+			error: self.into(),
+		}
+	}
```

Comment:
It looks like to me that we really should add a new function called `with_post_info`, so that callers can freely decide whether or not they want to use the default or modify any of the fields.

These methods should also exist in `PostDispatchInfo` rather than here as well, as it's not clear that the function actually returns a `DispatchErrorWtihPostInfo` after calling it.